tests/foreman/cli/test_reporttemplates.py

```diff
@@ -693,3 +694,70 @@ def test_positive_generate_with_name_and_org(self):
             host['name'],
             [item.split(',')[0] for item in result]
         )
+
+    @tier2
+    @pytest.mark.skip_if_open('BZ:1782807')
+    def test_positive_generate_ansible_template(self):
+        """ Report template named 'Ansible Inventory' (default name is specified in settings)
+        must be present in Satellite 6.7 and later in order to provide enhanced functionality
+        for Ansible Tower inventory synchronization with Satellite.
+
+        :id: f1f7adfc-9601-4498-95c8-3e82e2b36583
+
+        :setup:
+            1. A user with minimal required permissions: 'Ansible Tower Inventory Reader' role
+            2. A fake host to be checked in report output
+
+        :steps:
+            1. Check settings for default Ansible Inventory template name and ensure
+               the template is present
+            2. Try to render the template using the user with ATIR role
+            3. Check the fake host is present in the output
+
+        :expected result: Report template is present, renederable and provides output
+
+        :CaseImportance: Medium
+        """
+        settings = Settings.list({'search': 'name=ansible_inventory_template'})
+        assert 1 == len(settings)
+        template_name = settings[0]['value']
+
+        report_list = ReportTemplate.list()
+        assert template_name in [rt['name'] for rt in report_list]
+
+        login = gen_string('alpha').lower()
+        password = gen_string('alpha').lower()
+        loc = Location.info({'name': DEFAULT_LOC})
+        org = Org.info({'name': DEFAULT_ORG})
+
+        user = make_user({
+                'login': login,
+                'password': password,
+                'organization-ids': org['id'],
+                'location-ids': loc['id']
+        })
+
+        User.add_role({
+            'login': user['login'],
+            'role': 'Ansible Tower Inventory Reader',
```

Comment:
Usually we create new role, define new filter for that role with necessary permissions and use that. Probably for us to not be tied to specific string, which may change in the future.

I'm not saying you have to change this, I'm only pointing out difference from established pattern.

Reply:
I agree, I don't consider using constant strings in code as good practice too. However, in this case the referenced role is shipped as predefined role within Satellite to be used for rendering the template, so I would like to test this explicit role too. Does moving the string to the 'robottelo.constants' make sense?

Reply:
Nah, I would just leave it as is.
New role and filter makes it unnecessary to commit when existing role name change. If change of role name forces us to submit PR, then I don't see much difference between changing test itself and changing constants.
Plus constants turned out to be kitchen sink and adding one more random value used by single test won't really help anyone.